include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1051 @@
+
+                            float_X const weighting = particle[ weighting_ ];
+                            float3_X const mom = particle[ momentum_ ] / weighting;
+                            floatD_X const pos = particle[ position_ ];
+
+                            lcellId_t const cellIdx = particle[ localCellIdx_ ];
+                            DataSpace<simDim> const frameCellOffset(DataSpaceOperations<simDim>::template map<MappingDesc::SuperCellSize > (cellIdx));
+                            auto const localSupercellStart = (superCellIdx - mapper.getGuardingSuperCells()) * MappingDesc::SuperCellSize::toRT();
+                            int const index_y = frameCellOffset.y();
+                            auto const globalCellOffset = globalOffset
+                                                        + localSupercellStart
+                                                        + frameCellOffset;
+                            float_X const posX = ( float_X( globalCellOffset.x() ) + pos.x() ) * cellSize.x();
+
+                            atomicAdd( &(shCount_e[index_y]), weighting, ::alpaka::hierarchy::Threads{});
+                            //weighted sum of single Electron values (Momentum = particle_momentum/weighting)
+                            atomicAdd( &(shSumMom2[index_y]), mom.x() * mom.x() * weighting, ::alpaka::hierarchy::Threads{});
+                            atomicAdd( &(shSumPos2[index_y]), posX*posX*weighting, ::alpaka::hierarchy::Threads{});
+                            atomicAdd( &(shSumMomPos2[index_y]), mom.x()*posX* weighting, ::alpaka::hierarchy::Threads{});
+                        }
+                    }
+                );
+
+                // set frame to next particle frame
+                frame = pb.getPreviousFrame(frame);
+                forEachParticle(
+                    [&](
+                        uint32_t const linearIdx,
+                        uint32_t const idx
+                    )
+                    {
+                        /* Update particle for the next round.
+                         * The frame list is traversed from the last to the first frame.
+                         * Only the last frame can contain gaps therefore all following
+                         * frames are fully filled with particles.
+                         */
+                        currentParticleCtx[ idx ] = frame[ linearIdx ];
+                    }
+                );
+            }
+
+
+            // wait that all virtual threads updated the shared memory energies
+            __syncthreads( );
+
+            const int gOffset = ( (superCellIdx - mapper.getGuardingSuperCells()) * MappingDesc::SuperCellSize::toRT()).y();
+            ForEachIdx< SuperCellYDom >{ workerIdx }(
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    if( gOffset >= 0 && gOffset < subGridY - SuperCellSize::y::value ){
```

Comment:
Why do you subtract with  `SuperCellSize::y::value`. `gOffset` is the offset in cells relative to the begin of the gpu is can by definition not be lesser than zero and not larger than `subGridY` which is the global simulation size in y direction.

Please verify the if condition because currently it looks like it make no sense to have it here.